File: frontend/src/types/paper.ts
export interface Author {
  id: number;
  name: string;
  affiliation: string | null;
}

export interface PaperListItem {
  id: number;
  source: string;
  source_id: string;
  title: string;
  abstract: string | null;
  published_at: string | null;
  categories: string[] | null;
  keywords: string[] | null;
  status: string;
  authors: Author[];
}

export interface PaperDetail extends PaperListItem {
  summary: string | null;
  summary_zh: string | null;
  contributions: string[] | null;
  pdf_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface SubmitPaperRequest {
  source: "arxiv" | "semantic_scholar";
  paper_ids: string[];
}

export interface SubmitPaperResult {
  source_id: string;
  status: string;
  paper_id: number | null;
  message: string;
}

export interface SubmitPaperResponse {
  total: number;
  results: SubmitPaperResult[];
}

export interface ChatRequest {
  question: string;
  paper_ids?: number[];
}

export interface ChatResponse {
  answer: string;
  source_papers: number[];
  source_chunks: string[];
}

export type PaperStatus =
  | "pending"
  | "crawled"
  | "downloading"
  | "parsed"
  | "analyzed"
  | "embedded"
  | "ready";

File: src/daily_ai_papers/schemas/paper.py
"""Paper-related Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel


class AuthorResponse(BaseModel):
    id: int
    name: str
    affiliation: str | None = None

    model_config = {"from_attributes": True}


class PaperListItem(BaseModel):
    id: int
    source: str
    source_id: str
    title: str
    abstract: str | None = None
    published_at: datetime | None = None
    categories: list[str] | None = None
    keywords: list[str] | None = None
    status: str
    authors: list[AuthorResponse] = []

    model_config = {"from_attributes": True}


class PaperDetail(PaperListItem):
    summary: str | None = None
    summary_zh: str | None = None
    contributions: list[str] | None = None
    pdf_url: str | None = None
    created_at: datetime
    updated_at: datetime


class PaperSearchParams(BaseModel):
    query: str | None = None
    categories: list[str] | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None
    page: int = 1
    page_size: int = 20
